typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -385,6 +385,8 @@
          (if numeric-sc
              (apply or/sc numeric-sc (map t->sc non-numeric))
              (apply or/sc (map t->sc elems)))]
+        [(Intersection: elems)
+         (apply and/sc (for/list ([elem (in-set elems)]) (t->sc elem)))]
```

Comment:
Does this really work for complex intersections (for example, polymorphic types and others, or two different function types, etc)?